Automattic/studio
# Studio: Execute wp-cli inside Studio and receive its output

Closes https://github.com/Automattic/dotcom-forge/issues/7328

This PR adds the implementation to execute the `wp-cli` commands by pressing a button and receiving the result as a variable in React.  The approach in this PR adapts a solution from the [Playground solution ](https://github.com/WordPress/wordpress-playground/blob/157f581094fa7e031ccd16bf027bd25419f4dfa1/packages/playground/blueprints/src/lib/steps/wp-cli.ts#L50-L82).

## Testing Instructions



For testing purposes, I have added a textarea to add input, run CLI and receive the output on the frontend. The textarea is temporarily added under `Settings` tab in Studio and will be removed once the PR is approved. **I will only merge the implementation logic and not the frontend part**:

* Pull the changes from this branch locally
* Start the app with `nvm use && npm install && npm start`
* Create a site in Studio
* Navigate to Settings tab
* Scroll to the bottom of the page and observe the text area
* Add arguments such as `plugin list` (skip the `wp` part)
* Observe the output on the frontend listing all the plugins from the site
* Add an invalid argument such as `yoda` 
* Confirm that you get the error returned
* Try possible edge cases and different commands to see what is being output

https://github.com/Automattic/studio/assets/25575134/80744797-1943-4e1c-bcb3-eb0aec18767d

## Pre-merge Checklist
